ui/ts/pages/helpus.ts

```diff
@@ -0,0 +1,215 @@
+// source: pages/nodes.ts
+/// <reference path="../../bower_components/mithriljs/mithril.d.ts" />
+/// <reference path="../models/sqlquery.ts" />
+/// <reference path="../models/helpus.ts" />
+
+module AdminViews {
+  "use strict";
+
+  export module SubModules {
+    /**
+     * HelpUs is used for the sign up flow to capture user information
+     */
+    export module HelpUs {
+      import UserOptIn = Models.HelpUs.UserOptIn;
+      import showHelpUs = Models.HelpUs.showHelpUs;
+      import MithrilPromise = _mithril.MithrilPromise;
+
+      enum Stage {
+        Unknown,
+        // Show the "Help Us" banner at the top of the page.
+        Banner,
+        // Fade in the "Sign Up" modal.
+        ModalStart,
+        // Show the "Sign Up" modal.
+        Modal,
+        // Fade out the "Sign Up" modal if it's closed without signing up.
+        ModalClose,
+        // Transition the modal to the "Thanks" modal size while fading out the old text.
+        ThanksSize,
+        // Show the "Thanks" modal with the correct size and fade in the text.
+        ThanksText,
+        // Fade out the "Thanks" modal.
+        ThanksClose
+      }
+
+      class ModalController {
+        userData: UserOptIn = new UserOptIn();
+
+        // Current stage in the "Help Us" flow
+        stage: Stage = Stage.Unknown;
+        show: boolean = false;
+
+        // The closeOnEscape function, but bound to the correct `this`.
+        closeOnEscapeBound: (e: KeyboardEvent) => void;
+
+        constructor() {
+          // Adds an event listener for the escape key.
+
+          // Bind the closeOnEscape function.
+          this.closeOnEscapeBound = this.closeOnEscape.bind(this);
+          document.addEventListener("keydown", this.closeOnEscapeBound);
+
+          m.sync([
```

Comment:
Yeah "show help us" is making the full check here - checking if it's filled in or dismissed or if the flag isn't present. But since we're already pulling in the data for the ```UserOptIn``` object, we might as well just use that. I'll simplify it.